processor/batchprocessor/splitlogs_test.go

```diff
@@ -29,7 +29,11 @@ func TestSplitLogs_noop(t *testing.T) {
 	split := splitLogs(splitSize, td)
 	assert.Equal(t, td, split)
 
-	td.ResourceLogs().At(0).InstrumentationLibraryLogs().At(0).Logs().Resize(5)
+	i := 0
+	td.ResourceLogs().At(0).InstrumentationLibraryLogs().At(0).Logs().RemoveIf(func(_ pdata.LogRecord) bool {
```

Comment:
It is fine in the tests but for non-test code this may be much slower than Resize. Is there any non-test code here or in the contrib that may suffer from this change?

Reply:
I didn't find any non-test code in contrib that would need to use a pattern like this.